src/main/java/com/google/devtools/common/options/OptionsParserImpl.java

```diff
@@ -435,19 +470,25 @@ private OptionsParserImplResult parse(
 
       arg = swapShorthandAlias(arg);
 
-      if (containsSkippedPrefix(arg)) {
-        unparsedArgs.add(arg);
-        continue;
-      }
-
       if (arg.equals("--")) { // "--" means all remaining args aren't options
         Iterators.addAll(unparsedPostDoubleDashArgs, argsIterator);
         break;
       }
 
-      ParsedOptionDescription parsedOption =
-          identifyOptionAndPossibleArgument(
-              arg, argsIterator, priority, sourceFunction, implicitDependent, expandedFrom);
+      ParsedOptionDescription parsedOption;
+      if (containsSkippedPrefix(arg)) {
+        // Parse the skipped arg into a synthetic allowMultiple option to preserve its order
+        // relative to skipped args coming from expansions. Simply adding it to the residue would
+        // end up placing expanded skipped args after all explicitly given skipped args, which isn't
+        // correct.
+        parsedOption = ParsedOptionDescription.newParsedOptionDescription(skippedArgsDefinition,
```

Comment:
I also find the name not too great, but I would prefer not to change it in this PR: The rename would cause the diff to be much longer and less clean. If you want, I can submit a follow-up PR where we figure out and use a better name.